### DISCUSSION TITLE
New version with breaking changes

### DISCUSSION THREAD

**Discussion by gbishop** [UNC-Project-Open-AAC/OS-DPI#70] [Announcements]
The new version of our app has several important changes.

**These will break your existing designs. You'll need to reload the spreadsheet on the Content tab and you'll need to update any Grids or Radio buttons.**

## No tags

The concept of **Tags** has been replaced by more explicit and general **Filters**. This means there are no special columns in your spreadsheets. You can select the elements to appear in a grid using any combination of columns.

For example, suppose your content uses 2 columns as **Tags** in the old way. Your grid **Tags** entry might have been:
|Tags|
|------|
|'CHAT'|
|$topic|

In the new way your filters could be:

|Field|Operator|Value|
|------|------------|--------|
|#Role|equals|'CHAT'|
|#Topic|equals|$topic|

More to enter but more explicit and, I hope, easier to understand.

I have converted the *contact* example to the new format. We need to convert more.

## All sheets from your spreadsheet are now imported

This in combination with **No tags** allows much more flexible partitioning of your work. The various sheets in your spreadsheet don't have to share columns. Call them whatever you wish. The names *label*, *row*, *column*, *page*, *symbol* still serve special purposes but they don't need to line up across sheets. 

Furthermore the name of the sheet is available for testing with the **Filters** under the field name *sheetName*. If you test *sheetName* in your filters you'll have no worries of cross talk between your various sheets. 

## You can export the spreadsheet from the design

This should allow sharing designs as .osdpi files alone without having to also share the spreadsheet. The recipient can simply export the sheet from the design. The exported spreadsheet should include all the sheets from the original.

**Comment by sofia-bg** [comment]
This all sounds awesome! Thanks Gary
